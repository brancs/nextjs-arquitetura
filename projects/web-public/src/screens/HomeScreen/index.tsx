import Text from "@alura/design-system/components/Text";
import { sum } from "@alura/utils/math/sum";

export default function HomeScreen() {
  return (
    <>
      <Text tag={"h1"}>Alura - Home Page 2</Text>
      <Text tag={"p"}> Alura {sum(2, 4)}</Text>
    </>
  );
}
