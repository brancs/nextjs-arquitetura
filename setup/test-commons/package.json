{
  "name": "@alura/test-commons",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "files": [
    "base.ts"
  ],
  "scripts": {
    "lint": "echo 'Not necessary'",
    "test": "echo 'Not necessary'"
  },
  "devDependencies": {
    "@babel/core": "7.21.3",
    "@babel/preset-env": "7.20.2",
    "@babel/preset-react": "7.18.6",
    "@babel/preset-typescript": "7.21.0",
    "babel-jest": "29.5.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-styled-components": "7.1.1",
    "ts-node": "^10.9.1",
    "@testing-library/react": "14.0.0"
  }
}
