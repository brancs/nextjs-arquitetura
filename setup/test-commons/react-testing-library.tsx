import { render as rtlRender } from "@testing-library/react";

function DefaultWrapper({ children }: { children: any }) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  return (ui: any, options = {}) => {
    return rtlRender(ui, { wrapper: AllTheProviders, ...options });
  };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export const render = () =>
  console.error("Please create your own render method");
