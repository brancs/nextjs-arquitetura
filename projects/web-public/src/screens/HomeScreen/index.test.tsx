import { ReactNode } from "react";
import HomeScreen from ".";
import { customRender } from "@alura/test-commons/react-testing-library";

function CustomWrapper({ children }: { children: ReactNode }) {
  return <section>{children}</section>;
}

const render = customRender(CustomWrapper);

describe("<HomeScreen />", () => {
  it("renders the page", () => {
    const { container } = render(<HomeScreen />);
    expect(container).toMatchSnapshot();
  });
});
