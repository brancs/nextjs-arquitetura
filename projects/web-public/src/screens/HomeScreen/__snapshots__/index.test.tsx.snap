// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders the page 1`] = `
.c0 {
  font-family: sans-serif;
}

<div>
  <section>
    <h1
      class="c0"
    >
      Alura - Home Page 2
    </h1>
    <p
      class="c0"
    >
       Alura 
      6
    </p>
  </section>
</div>
`;
